clang/lib/Serialization/ASTReader.cpp

```diff
@@ -10079,6 +10079,11 @@ void ASTReader::finishPendingActions() {
     }
     PendingDeducedVarTypes.clear();
 
+    // Load the delayed preferred name attributes.
+    for (unsigned I = 0; I != PendingDeferredAttributes.size(); ++I)
+      loadDeferredAttribute(PendingDeferredAttributes[I]);
+    PendingDeferredAttributes.clear();
```

Comment:
```suggestion
    // Load the delayed preferred name attributes.
    while (!PendingDeferredAttributes.empty()) {
      auto DeferredAttributes = std::move(PendingDeferredAttributes);
      for (DeferredAttribute &DA : DeferredAttribute)
        loadDeferredAttribute(DA);
    }
```

Technically, it is possible to update `PendingDeferredAttributes` during `loadDeferredAttribute()`.

Reply:
Thanks. The original snippet seems more concise though, so it might be preferable to stick to it for now.